subprojects/core-api/src/main/java/org/gradle/api/tasks/TaskContainer.java

```diff
@@ -73,6 +73,18 @@
     @Incubating
     <T extends Task> Provider<T> getByNameLater(Class<T> type, String name) throws InvalidUserDataException;
 
+    /**
+     * Configures the named element in this collection using the given action. Actions are run in the order added.
+     *
+     * <strong>Note: this method currently has a placeholder name and will almost certainly be renamed.</strong>
+     *
+     * @param name Name of the element to be configured
+     * @param action A {@link Action} that can configure the element when required.
+     * @since 4.8
+     */
+    @Incubating
+    void configureLater(String name, Action<? super Task> action);
```

Comment:
I think we should go with an API more in line with what we have on `BinaryContainer` where element lookup and configuration are somewhat separated. Instead of this method, we'd add a `configure {}` method to the provider that is returned from `getByNameLater()` and `createLater()`. This allow us to later add other kinds of "get a single task that matches this constraint" lookup methods without needing a configure method for each of these as well (they would just return the same provider subtype) and we can add other lifecycle hooks and control methods to the provider subtype without needing a bunch of overloaded methods on `TaskContainer` either.

Reply:
I removed this method from the API and added a `TaskProvider` and `TaskProvider.configure(Action)`.